pkg/util/port/portlist/poller_windows.go

```diff
@@ -18,21 +16,86 @@ var ErrNotImplemented = errors.New("not implemented yet")
 
 // init initializes the Poller by ensuring it has an underlying
 func (p *Poller) init() {
-	p.os = newWindowsOSImpl(p.IncludeLocalhost)
+	p.os = newWindowsImpl(p.IncludeLocalhost)
 }
 
-type windowsOSImpl struct {
+type windowsImpl struct {
+	known            map[famPort]*portMeta
 	includeLocalhost bool
 }
 
-func newWindowsOSImpl(includeLocalhost bool) osImpl {
-	return &windowsOSImpl{
+type famPort struct {
+	proto string
+	port  uint16
+	pid   uint32
+}
+
+type portMeta struct {
+	port Port
+	keep bool
+}
+
+func newWindowsImpl(includeLocalhost bool) osImpl {
+	return &windowsImpl{
+		known:            map[famPort]*portMeta{},
 		includeLocalhost: includeLocalhost,
 	}
 }
+func (*windowsImpl) Close() error { return nil }
 
-func (im *windowsOSImpl) AppendListeningPorts(_ []Port) ([]Port, error) {
-	return nil, ErrNotImplemented
-}
+func (im *windowsImpl) AppendListeningPorts(base []Port) ([]Port, error) {
+	tab, err := GetConnTable()
+	if err != nil {
+		return nil, err
+	}
 
-func (*windowsOSImpl) Close() error { return ErrNotImplemented }
+	for _, pm := range im.known {
+		pm.keep = false
+	}
+
+	ret := base
+	for _, e := range tab.Entries {
+		if e.State != "LISTEN" {
```

Comment:
I wonder if the port is not "CLOSED" it probably would also conflict in attempt to open it, no?